packages/code-du-travail-frontend/src/outils/ConventionCollective/common/TrackingContext.tsx

```diff
@@ -0,0 +1,50 @@
+import pDebounce from "p-debounce";
+import React, { createContext, useContext } from "react";
+import { v4 as generateUUID } from "uuid";
+
+import { matopush } from "../../../piwik";
+
+type TrackingContext = {
+  trackEvent: (
+    category: string,
+    action: string,
+    name?: string,
+    value?: string
+  ) => void;
+  uuid: string;
+  title: string;
+};
+
+export const trackingContext = createContext<TrackingContext>({
+  title: null,
+  trackEvent: null,
+  uuid: null,
+});
+
+export function useTrackingContext(): TrackingContext {
+  return useContext(trackingContext);
+}
+
+type Props = {
+  uuid?: string;
+  title: string;
+  children: React.ReactNode;
+};
+
+const { Provider } = trackingContext;
+
+export function TrackingProvider({
+  title,
+  children,
+  uuid = generateUUID(),
+}: Props): JSX.Element {
+  const trackEvent = (...params) => {
+    matopush(["trackEvent", ...params]);
+  };
+  const debuncedTrackEvent = pDebounce(trackEvent, 2000);
```

Comment:
Misspelling `debouncedTrackEvent` 
Also, why we debounce the trackEvent?

Reply:
It was too avoid too much search event (z, za, zar, zara) but it could lead to issue if user complete search / select an agreement under 2s